lxd/storage/drivers/clients/alletra_wsapi1.go

```diff
@@ -525,3 +579,91 @@ func (p *AlletraClient) DeleteHost(hostName string) error {
 func (p *AlletraClient) UpdateHost(hostName string, qns []string) error {
 	return fmt.Errorf("Failed to update host %q. Operation not supported.", hostName)
 }
+
+func (p *AlletraClient) createVolume(poolName string, volName string, cpg string, sizeBytes int64) error {
+	req := map[string]any{
+		"name":    volName,
+		"cpg":     cpg,
+		"sizeMiB": sizeBytes / 1024 / 1024,
+		"tpvv":    true, // thinly provisioned volume
+	}
+
+	url := api.NewURL().Path("api", "v1", "volumes")
+	err := p.requestAuthenticated(http.MethodPost, url.URL, req, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to create volume %q in storage pool %q: %w", volName, poolName, err)
+	}
+
+	return nil
+}
+
+// CreateVolume creates a new volume in the given storage pool. The volume is created with
+// supplied size in bytes. Upon successful creation, volume's ID is returned.
+func (p *AlletraClient) CreateVolume(poolName string, volName string, cpg string, sizeBytes int64) error {
+	err := p.createVolume(poolName, volName, cpg, sizeBytes)
+	if err != nil {
+		return err
+	}
+
+	// Add a newly created volume to a volume set
+	err = p.modifyVolumeSet(poolName, apiVolumeSetMemberAdd, volName)
+	return err
+}
+
+// GetVolume returns the volume for a given volName.
+func (p *AlletraClient) GetVolume(poolName string, volName string) (*hpeVolume, error) {
+	var resp hpeVolume
+
+	url := api.NewURL().Path("api", "v1", "volumes", volName)
+	err := p.requestAuthenticated(http.MethodGet, url.URL, nil, &resp)
+	if err != nil {
+		hpeErr, ok := err.(*hpeError)
+		if ok {
+			switch hpeErr.Code {
+			case apiErrorNonExistentVol:
+				return nil, api.StatusErrorf(http.StatusNotFound, "Volume/snapshot %q not found", volName)
```

Comment:
What does "volume/snapshot" mean in this context, is it "volume or snapshot" or "snapshot of volume"?

Reply:
Error message was improved.